assets/src/dashboard/components/toaster/provider.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { createContext, useMemo, useCallback, useState } from 'react';
+import PropTypes from 'prop-types';
+
+export const ToasterContext = createContext(null);
+
+const ToastProvider = ({ children }) => {
+  const [toasts, setToasts] = useState({});
+  const activeToasts = useMemo(
+    () => Object.values(toasts).filter((toast) => toast.isActive),
+    [toasts]
+  );
+
+  const removeToast = useCallback(
+    (id) =>
+      id &&
+      setToasts((existingToasts) => ({
+        ...existingToasts,
+        [id]: { ...existingToasts[id], isActive: false },
```

Comment:
After they animate out do we want to completely remove them from the array?

Reply:
@BrittanyIRL if you wanted to do this, you could expose a delete method from the provider and call it with 
```
onTransitionEnd={() => !isActive && deleteToast(id)}
```
 on the individual toasts. or `onAnimationEnd` if it's using `animation` instead of `transition`

Reply:
nope! we really don't :D 

this component has really evolved and after swapping up from using arrays to using an object and memoizing and getting storybook in working order (which i think is the real reason why I was keeping toasts around after being dismissed) and I just missed this as an update completely. Thank you. Updated and retested.  

Love how clean this thing turned out!